core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/MethodConfigurationComponent.scala

```diff
@@ -188,12 +188,28 @@ trait MethodConfigurationComponent {
       }
     }
 
+    object MethodConfigurationCloneQuery extends RawSqlQuery {
+      val driver: JdbcProfile = MethodConfigurationComponent.this.driver
+
+      def cloneMethodConfigs(srcWsId: UUID, destWsId: UUID): WriteAction[Int] = {
+
+        sqlu"""insert into METHOD_CONFIG (NAMESPACE, NAME, WORKSPACE_ID, ROOT_ENTITY_TYPE, METHOD_URI,
+               DELETED, METHOD_CONFIG_VERSION, DATA_REFERENCE_NAME) select NAMESPACE, NAME,
+               $destWsId, ROOT_ENTITY_TYPE, METHOD_URI, 0, METHOD_CONFIG_VERSION,
+               DATA_REFERENCE_NAME from METHOD_CONFIG where WORKSPACE_ID=$srcWsId and deleted = 0"""
+      }
+    }
```

Comment:
Awesome on seeing the opportunity here and forging ahead with optimizing this too!

Unfortunately I don't think this query is sufficient. It is failing to copy rows from the `METHOD_CONFIG_INPUT` and `METHOD_CONFIG_OUTPUT` tables.

My suggestion is to remove the method config optimization from this PR and tackle it in a follow-on, to unblock merging of the entity optimizations - but if you'd rather tackle it all in one, great.

Have you looked at test coverage for this? (I don't know much about method config coverage)